### PR TITLE
Added Yokai namespace packages

### PR DESCRIPTION
**Pull request by yann-eugone**
| Q             | A
| ------------- | ---
| License       | MIT

### CODE REVIEW

**Comment by yann-eugone** [comment]
Packages may be more than simply a bundle : library with symfony Bundle integration.

As I see, with ApiPlatform, there is no problem with this

- [ApiPlatform recipe](https://github.com/symfony/recipes/blob/master/api-platform/core/2.1/manifest.json#L3)
- [ApiPlatform package type](https://github.com/api-platform/core/blob/2.1/composer.json#L3)

Do someone knows why it is required that the composer package is declared **as** a Symfony bundle ?

**Comment by maxhelias** [comment]
I think this is not a requirement. It's for Flex in the case where there is no recipe, it will try to create one via the definition of the autoload with the `composer.json`

Can you confirm @Nyholm ?

**Comment by Nyholm** [comment]
Yeah, That is correct.

File: yokai/messenger-bundle/1.0/config/packages/yokai_messenger.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+yokai_messenger:
+    channels:
+        # Each channel has its own global configuration
+        swiftmailer:
```
Comment (Nyholm) [inline yokai/messenger-bundle/1.0/config/packages/yokai_messenger.yaml]: Is this the same as default configuration?

File: yokai/enum-bundle/2.2/manifest.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "bundles": {
+        "Yokai\\EnumBundle\\YokaiEnumBundle": ["all"]
```
Comment (Nyholm) [inline yokai/enum-bundle/2.2/manifest.json]: Flex is smart enough to install this bundle without a recipe. Just make sure to specify "symfony-bundle" as a type in composer.json

File: yokai/messenger-bundle/1.0/config/packages/yokai_messenger.yaml (hunk 1 above)
Comment (yann-eugone) [reply yokai/messenger-bundle/1.0/config/packages/yokai_messenger.yaml]: The whole file is not the default configuration : because of the example, but yes, channels are all disabled by default (usage of `canBeEnabled` config helper)

File: yokai/enum-bundle/2.2/manifest.json (hunk 2 above)
Comment (yann-eugone) [reply yokai/enum-bundle/2.2/manifest.json]: I understand that. In the next major version of this repository, it will become less coupled to symfony, and transform to a library.

As for api platform, I was expecting to distribute everything in the same repository : the library, the symfony bridge, ...

I was expecting that i can add a bundle recipe for a package that is not marked as a bundle explicitely. But regarding the automated tests, I believe it is not possible. Do you confirm ?

Comment (Nyholm) [reply yokai/enum-bundle/2.2/manifest.json]: > I was expecting that i can add a bundle recipe for a package that is not marked as a bundle explicitely.

You can, but that is an edge case. 

> But regarding the automated tests, I believe it is not possible. Do you confirm ?

Im not sure what you mean? The review bot will not be happy if you submit a `library`, that is correct.

**Comment by Nyholm** [comment]
@symfony-flex-server review please

**Comment by Nyholm** [comment]
I cannot approve this unless the review bot is happy. For example. You can never depend on `symfony/symfony`

**Comment by yann-eugone** [comment]
closing. will split this in several PRs this next week
